connector/countconnector/config_integration_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+package countconnector
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/connector/connectortest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/otelcol/otelcoltest"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/countconnector/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
+)
+
+// Ensure basic count connector configs produce expected metrics.
+// See ./testdata/config-*.yaml for specifics.
+// Test cases:
+// 1) `count:` with empty subconfig uses defaults.
+// 2) `count::logs:` with empty subconfig uses defaults.
+// 3) `count::logs::<custom metric>` overrides default.
+
+func newCountConfigFromYAML(t *testing.T, yamlFile string) *Config {
+	t.Helper()
+
+	factories, err := otelcoltest.NopFactories()
+	require.NoError(t, err)
+	factories.Connectors[metadata.Type] = NewFactory()
+
+	cfg, err := otelcoltest.LoadConfigAndValidate(filepath.Join("testdata", yamlFile), factories)
+	require.NoError(t, err)
+
+	ccfg, ok := cfg.Connectors[component.NewID(metadata.Type)].(*Config)
+	require.True(t, ok)
+	require.NotNil(t, ccfg)
+
+	require.NoError(t, ccfg.Validate())
```

Comment:
This is the same check as what's done in `otelcoltest.LoadConfigAndValidate`, isn't it? (We can remove if so)